### PR TITLE
Adds a presubmit check for `window.document`

### PR DESCRIPTION
**Pull request by prateekbh**
- `window.document` fails when an extension is used with shadow runtime. This adds a presubmit check for the same
- Fixes Task 1  in #19514

### CODE REVIEW

File: build-system/tasks/presubmit-checks.js [hunk 1]
```diff
@@ -42,6 +42,9 @@ const backwardCompat = 'This method must not be called. It is only retained ' +
 const realiasGetMode = 'Do not re-alias getMode or its return so it can be ' +
     'DCE\'d. Use explicitly like "getMode().localDev" instead.';
 
+const shadowDocUnsafe = 'Do not use this API as this is not available in' +
```
Comment (choumx) [inline build-system/tasks/presubmit-checks.js]: Needs space between "in" and "shadow".

File: build-system/tasks/presubmit-checks.js [hunk 2]
```diff
@@ -654,6 +657,20 @@ const forbiddenTerms = {
       'src/inabox/amp-inabox-lite.js',
     ],
   },
+  'window\.document': {
```
Comment (choumx) [inline build-system/tasks/presubmit-checks.js]: Simple string match linter is insufficient here. E.g. in #19511 the problematic snippet was `this.win.document`.

Have you tried doing this in conformance-config.textproto?

**Comment by prateekbh** [comment]
@choumx i just came across https://github.com/ampproject/amphtml/blob/master/extensions/amp-access/0.1/amp-access-iframe.js#L80

Is this not problematic in the same way as #19511 was?

**Comment by prateekbh** [comment]
I tried 
```
requirement: {
  type: BANNED_NAME
  error_message: 'Document should never be accessed directly, please use this.getAmpDoc()'
  value: 'window.document'
}
```

but it couldn't detect the defect in #19511.
Is there a better way?

Can a custom lint rule help here instead?

**Comment by choumx** [comment]
`document.createElement()` is fine because it's independent of DOM scoping. By contrast, waiting for `document.body` is not.

**Comment by mrjoro** [comment]
Is this PR still active?

**Comment by prateekbh** [comment]
No, never found a reliable way of covering all the edge cases

**Pull request closed without merging**